### ISSUE TITLE
Request for help with structure of Objects365 dataset for validation

### ISSUE THREAD

**Issue by gd1925** [ultralytics/yolov5#9223]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi all,

I am trying to get the mAP scores for yolov5 model trained on Objects365 as found in this link - [https://github.com/ultralytics/yolov5/releases/download/v6.0/yolov5m_Objects365.pt](url)
I wanted to calculate mAP scores for this model using the val.py. I already have the dataset with me and I wanted to ask if its possible to help me with the structure it needs for the dataset to be in, in order to run the code? I keep running into the following error:  (0 images found) 

![image](https://user-images.githubusercontent.com/50822928/187417301-ac9bc521-4195-4a41-a4cf-486bd99be221.png)

**Comment by glenn-jocher** [comment]
@gd1925 Objects 365 usage and download instructions are in the yaml:
https://github.com/ultralytics/yolov5/blob/5fb267f3e5dc86675d508e1b08d20fc0e2e84003/data/Objects365.yaml#L1-L8

Dataset will autodownload on first use. Your data must be in YOLOv5 format. Please see our [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) tutorial for full documentation on dataset setup and all steps required to start training your first model. A few excerpts from the tutorial:

### 1.1 Create dataset.yaml

[COCO128](https://www.kaggle.com/datasets/ultralytics/coco128) is an example small tutorial dataset composed of the first 128 images in [COCO](http://cocodataset.org/#home) train2017. These same 128 images are used for both training and validation to verify our training pipeline is capable of overfitting. [data/coco128.yaml](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), shown below, is the dataset config file that defines 1) the dataset root directory `path` and relative paths to `train` / `val` / `test` image directories (or *.txt files with image paths) and 2) a class `names` dictionary:
```yaml
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes (80 COCO classes)
names:
  0: person
  1: bicycle
  2: car
  ...
  77: teddy bear
  78: hair drier
  79: toothbrush

```


### 1.2 Create Labels

After using a tool like [Roboflow Annotate](https://roboflow.com/annotate?ref=ultralytics) to label your images, export your labels to **YOLO format**, with one `*.txt` file per image (if no objects in image, no `*.txt` file is required). The `*.txt` file specifications are:

- One row per object
- Each row is `class x_center y_center width height` format.
- Box coordinates must be in **normalized xywh** format (from 0 - 1). If your boxes are in pixels, divide `x_center` and `width` by image width, and `y_center` and `height` by image height.
- Class numbers are zero-indexed (start from 0).

<p align="center"><img width="750" src="https://user-images.githubusercontent.com/26833433/91506361-c7965000-e886-11ea-8291-c72b98c25eec.jpg"></p>

The label file corresponding to the above image contains 2 persons (class `0`) and a tie (class `27`):

<p align="center"><img width="428" src="https://user-images.githubusercontent.com/26833433/112467037-d2568c00-8d66-11eb-8796-55402ac0d62f.png"></p>


### 1.3 Organize Directories

Organize your train and val images and labels according to the example below. YOLOv5 assumes  `/coco128` is inside a `/datasets` directory **next to** the `/yolov5` directory. **YOLOv5 locates labels automatically for each image** by replacing the last instance of `/images/` in each image path with `/labels/`. For example:
```bash
../datasets/coco128/images/im0.jpg  # image
../datasets/coco128/labels/im0.txt  # label
```

<p align="center"><img width="700" src="https://user-images.githubusercontent.com/26833433/134436012-65111ad1-9541-4853-81a6-f19a3468b75f.png"></p>
</details>

Good luck 🍀 and let us know if you have any other questions!

**Comment by gd1925** [comment]
Hi @glenn-jocher,

Thank you so much for your reply. I have already downloaded the dataset so I was thinking of arranging the val folders and train folders in the way that the val.py could directly run without the need of  downloading again. Also I have been facing proxy connection issue and I have been unable to run the val.py directly. Is it possible to get a reference on the structure of the folders? 

Thank you for your continued help and support!

**Comment by glenn-jocher** [comment]
@gd1925 my earlier message in https://github.com/ultralytics/yolov5/issues/9223#issuecomment-1231559616 already explains the required folder structure, see that.
